stat/cca_example_test.go

```diff
@@ -85,16 +85,17 @@ func ExampleCC() {
 	}
 
 	// Unpack cc.
+	var pVecs, qVecs, phiVs, psiVs mat.Dense
 	ccors := cc.CorrsTo(nil)
```

Comment:
I vaguely recall that we discussed the slice case and decided that the utility of the `range fn(nil)` case was worth keeping. Do I remember correctly?

Reply:
I do not remember that at all (though it could have happened). We talked about it in `floats`, but that's a bit different because `floats` does not currently allow `nil` inputs. I think it's good to keep the nil case for slices, since we do that in a lot of places, and it's not that common to use pointers to slices. I think the default should be "you can either supply the exact size, or nil". Matrices end up (should be) the special case, because we have to use receivers, and we want those to be able to be Empty. Supplying empty matrices as destinations is just a follow-on case to that

Reply:
As an addendum, this is going to be used as

```
dst := &mat.Dense{} or dst.Reset()
a,b := mds.TorgersonScaling(dst, stuff)
```
and so either way it seems pretty obvious to me that `dst` can't be supplying any useful information to the function.

Reply:
I think this second comment belonged above.

Reply:
yes it did, sorry.